lib/susedistribution.pm

```diff
@@ -631,7 +631,7 @@ Make sure the right user is logged in, e.g. when using remote shells
 =cut
 sub ensure_user {
     my ($user) = @_;
-    enter_cmd("su - $user") if $user ne 'root';
+    enter_cmd("test \$(id -un) == $user || su - $user") if $user ne 'root';
```

Comment:
Thinking twice on it, we might need to add quotes to avoid globbing in the username :)

```suggestion
    enter_cmd(sprintf('test "$(id -un)" == "%s" || su - "%s"', $user, $user)) if $user ne 'root';
```
WDYT?